rfc/2022/2022-08-23_CIT_hostnames.md

```diff
@@ -0,0 +1,227 @@
+  - Outstanding issues are fixed and re-deployed. Iterate as needed until Staging is stable.
+  - Once Staging is confirmed to be stable, prepare the Production deployment.
+
+
+**Step 4** (Coordinated release of vets-api and vets-website repositories in Production)
+  - Broadcast an internal notification that Production may be disrupted.
+  - Enable a sitewide, public-facing downtime notification informing users that Sign In will be unavailable.
+  - Site Reliability Engineering notified of impending deployment to Production.
+  - Infrastructure On-Call Team notified of impending deployment to Production.
+  - In parallel:
+    - The Identity Team prepares a PR for updates to the vets-api authentication URLs for Production.
+    - The IAM team updates the API callback URLs.
+  - Release Tools Team prepares a PR for updates to globally configured API URL for vets-website for Production.
+  - The scheduled daily auto-deploy is disabled for vets-website and vets-api so that the deployment process can be fully controlled.
+  - Identity Team manually runs only the build and release steps for vets-api.
+  - Release Tools Team triggers vets-website deploy manually. The built-in pause in the manual release will need to be approved to resume the deployment after the Build and Release steps are finished.
+  - Based on Pre-Release coordination research, trigger the deployment of both vets-api and vets-website so that they finish at approximately the same time.
+  - All teams QA/UAT Production for issues.  
+![staggered-deploy](https://user-images.githubusercontent.com/30317/186273089-4d101052-0166-4fde-aef5-d29214b866b2.png)
+
+
+### Post-Release Tasks
+
+#### **Site Reliability Engineering**
+  - SRE playbooks updated to use the *platform-api.va.gov hostname.
+
+
+#### **Release Tools Team**
+  - Search and replace documented instances of  `api.va.gov` API references with `platform-api.va.gov`, e.g., README files, script files, code comments, etc.
+
+
+#### **Console Services**
+  - Adds linting rules or triggered Github action to prevent use of the *api.va.gov hostname.
+
+
+#### **Mobile Flagship App**
+Because the Mobile Flagship App is not bound by the same authentication workflows as vets-api and vets-website, it can be deployed separately.
+  - The global configuration variables for new versions of the Mobile Flagship app are edited to use *platform-api.va.gov hostname. 
+  - The global configuration variables for legacy versions of the Mobile Flagship app will continue to use the older api.va.gov hostname and will be routed via the Lighthouse gateway.
+
+
+#### **Lighthouse**
+At this time the Lighthouse team is working to transition to using the new Apigee Gateway (timeline TBD.) Until that time, Kong will continue to serve as its API gateway and will honor requests to `api.va.gov/services/*` and `api.va.gov/internal/*`.
+  - Once implemented, the new Apigee gateway will continue to honor legacy/mobile app requests to `api.va.gov/mobile/*`.
```

Comment:
What is the route `api.va.gov/mobile/*` use for? If this is the VA Flagship mobile app then it should already be pointing at `platform-api.va.gov` by the time migration to Apigee occurs.

Reply:
@mchelen-gov, the `api.va.gov/mobile/*` endpoints are to support older versions of the mobile app that some people will have installed on their phones. The VA Flagship Mobile team requested this accommodation from Lighthouse, so as to prevent forcing people to upgrade. That being said, newer versions of the mobile app will definitely point at `*platform-api.va.gov`.